### ISSUE TITLE
How to use external class?

### ISSUE THREAD

**Issue by mkarg** [sonatype/modello#24]
I want to do this...
```java
<field>
   <name>fileMappers</name>
   <version>2.0.0+</version>
   <association>
      <type>org.codehaus.plexus.components.io.filemappers.FileMapper</type>
   </association>
   <description>
     <![CDATA[
     Allows to ...
     ]]>
   </description>
</field>
```
...but apparently this is wrong, because it cannot find the class:
```bash
 There is no class 'org.codehaus.plexus.components.io.filemappers.FileMapper' in the version range '2.0.0+'.
```
It seems I have to somehow make known existing classes to the model, but how?

**How to do that _correct_?**

**Comment by mkarg** [comment]
@plamentotev If you can answer this question I will be able to extend Maven Assembly Plugin in the same way I did it with the Maven Dependency Plugin. :-)

**Comment by plamentotev** [comment]
@mkarg I'll look at it but this repo is not the current one. The project moved here - https://github.com/codehaus-plexus/modello Unfortunately that causes a lot of confusion.

**Comment by plamentotev** [comment]
@mkarg I don't think you can reference type that is defined outside the model. What you can do is define your own class and use it after that to configure the file mappers. That is what Apache Maven File Management API does - https://github.com/apache/maven-file-management/blob/master/src/main/java/org/apache/maven/shared/model/fileset/mappers/MapperUtil.java#L91

**Comment by mkarg** [comment]
@plamentotev Introducing a class just for the sake of workaround a modello shortcoming? That is really ugly. The Assembly Plugin authors shouldn't have used Modello. ;-(

**Comment by mkarg** [comment]
@plamentotev Thank you for your kind help. The needed solution is such ugly that I reject to implement file mappers in assembly plugin. If one find days modello learns to simply add imports I will be happy to continue. Until then, I will stick with the dependency plugin. BTW, did you find time for a review? :-)
